### ISSUE TITLE
Hi @JoaoLopesFerreira99,

### ISSUE THREAD

**Issue by vikash-48413** [ultralytics/ultralytics#15957]
Hi @JoaoLopesFerreira99,

Thank you for reporting this issue and providing the detailed traceback and code snippet. It looks like you're encountering an `AttributeError` related to the `v10DetectLoss` attribute when attempting to load your YOLOv10 model.

To help us diagnose and resolve this issue more effectively, could you please verify the following:

1. **Ensure Latest Version**: Make sure you are using the latest version of the Ultralytics package. You can update it using:
    ```bash
    pip install --upgrade ultralytics
    ```

2. **Reproducible Example**: It's crucial to have a minimal reproducible example to pinpoint the problem accurately. Your provided code is a great start. Please ensure that the issue persists with the latest package version and provide any additional context if necessary. More details on creating a minimal reproducible example can be found [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

3. **Model Compatibility**: Verify that the model file `YOLOv10n.pt` is compatible with the version of the Ultralytics package you are using. Sometimes, models trained or saved with different versions might have compatibility issues.

Here's a slightly modified version of your script to ensure it aligns with the latest package updates:

```python
import torch
import ultralytics

def main():
    # Print the version of Ultralytics
    print(ultralytics.__version__)

    # Load the trained model
    model = ultralytics.YOLO('C:/Users/JFerreira/Desktop/Models/_testing/models/YOLOv10n.pt')

    # Dataset path
    dataset_path = 'C:/Users/JFerreira/Desktop/Models/_testing/test_dataset/airsim.yaml'

    # Run inference on the dataset and gather metrics
    model.val(data=dataset_path)

if __name__ == '__main__':
    main()
```

If the issue persists after these steps, please let us know. Providing any additional logs or context will be helpful. We appreciate your cooperation and look forward to resolving this issue promptly.

_Originally posted by @glenn-jocher in https://github.com/ultralytics/ultralytics/issues/14170#issuecomment-2205001147_

**Comment by vikash-48413** [comment]
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-22-3a80d0b23ae4>](https://localhost:8080/#) in <cell line: 7>()
      5 
      6 # Load the YOLOv10n model with the custom trained weights
----> 7 model = YOLO('/content/drive/MyDrive/CAR Parking /CAR-6000/Yolo-V10n-6000-car/weights/best.pt')
      8 
      9 # Path to the input video


how to resolve it
8 frames
[/usr/local/lib/python3.10/dist-packages/torch/serialization.py](https://localhost:8080/#) in find_class(self, mod_name, name)
   1513                     pass
   1514             mod_name = load_module_mapping.get(mod_name, mod_name)
-> 1515             return super().find_class(mod_name, name)
   1516 
   1517     # Load the data (which may in turn use `persistent_load` to load tensors)

AttributeError: Can't get attribute 'v10DetectLoss' on <module 'ultralytics.utils.loss' from '/usr/local/lib/python3.10/dist-packages/ultralytics/utils/loss.py'>

**Comment by Y-T-G** [comment]
It happens because you're loading a YOLOv10 model trained with the [YOLOv10 repo](https://github.com/THU-MIG/yolov10) with ultralytics.

**Comment by vikash-48413** [comment]
I have trained a YOLO v10n model on a car dataset. I need to perform
detection and then tracking of cars in a VS Code environment. However, when
I try to export the model to ONNX format, I encounter an error related to
detection loss. How can I resolve this issue? Do I need to retrain the
model, or is there another solution or alternative?

**Comment by Y-T-G** [comment]
Yes, you need to retrain if you wish to use official ultralytics.
